interfaces/ITextToSpeech.h

```diff
@@ -0,0 +1,116 @@
+
+            virtual ~INotification() {}
+            // Signal changes on the subscribed namespace
+            // @brief Notify TTS enabled/disabled
+            // @param state enabled/disabled
+            virtual void Enabled(const bool &state) = 0;
+            // @brief Notify change in voice used for speaking
+            // @param voice voice changed 
+            virtual void VoiceChanged(const string &voice) = 0;
+            // @brief Notify speechid based on the speech state(eg: start,pause,..etc)
+            // @param speechid id of the text
+            virtual void WillSpeak(const uint32_t &speechid) = 0;
+            virtual void SpeechStart(const uint32_t &speechid) = 0;
+            virtual void SpeechPause(const uint32_t &speechid) = 0;
+            virtual void SpeechResume(const uint32_t &speechid) = 0;
+            virtual void SpeechInterrupted(const uint32_t &speechid) = 0;
+            virtual void NetworkError(const uint32_t &speechid) = 0;
+            virtual void PlaybackError(const uint32_t &speechid) = 0;
+            virtual void SpeechComplete(const uint32_t &speechid) = 0;
+        };
+
+        virtual ~ITextToSpeech() {}
+
+        virtual void Register(ITextToSpeech::INotification* sink) = 0;
+        virtual void Unregister(ITextToSpeech::INotification* sink) = 0;
+        // @property
+        // @brief Query the status/enable tts
+        // @param enable status/enable
+        virtual uint32_t Enable(const bool enable) = 0;
+        virtual uint32_t Enable(bool &enable /* @out */) const = 0;
+        // @property
+        // @brief Set the tts configuration attributes
+        // @param config tts configuration
+        // @param status return status
+        virtual uint32_t SetConfiguration(const Configuration &config,TTSErrorDetail &status/* @out */) = 0;
+        virtual uint32_t SetFallbackText(const string &scenario,const string &value) = 0;
+        virtual uint32_t SetAPIKey(const string &apikey) = 0;
+        virtual uint32_t SetPrimaryVolDuck(const uint8_t &prim) = 0;
+        // @property
+        // @brief Retrieve tts configuration attributes 
+        // @param config tts configuration
+        virtual uint32_t GetConfiguration(Configuration &config/* @out */) = 0;
+        // @property
+        // @brief List voices available 
+        // @param language input
+        // @param voices list of voices
+        virtual uint32_t ListVoices(const string &language,RPC::IStringIterator*& voices/* @out */) = 0;
+        // @property
+        // @brief Speaks text provided
+        // @param text for conversion
+        // @param speechid returns id for the text 
+        // @param status return status
+        virtual uint32_t Speak(const string &text,uint32_t &speechid/* @out */,TTSErrorDetail &status/* @out */) = 0;
+        // @property
+        // @brief Cancel the speech 
+        // @param speechid id of text to be cancelled 
+        virtual uint32_t Cancel(const uint32_t &speechid) = 0;
```

Comment:
It looks (as this is a const uint32_t) that this does not need to be a reference. So the reference and the const contradict each other. I assume it should be "virtual uint32_t Cancel(const uint32_t speechid)". This allows the generators to optimize this to only marhalling code outbound and not to expect something inbound.